hs/src/Reach/Eval.hs

```diff
@@ -353,6 +353,7 @@ makeEnv cns = do
   e_droppedAsserts <- newCounter 0
   let e_appr = Left $ app_default_opts e_id e_droppedAsserts cns
   let e_compileProg = const $ impossible "compileProg"
+  e_universe <- newCounter uni
```

Comment:
It is important that this is a global counter. I think that `makeEnv` should take a `uniC :: Counter` and you should do `e_universe <- readCounter uniC`